packages/guides/src/Compiler/DocumentNodeTraverser.php

```diff
@@ -50,7 +50,7 @@ private function traverseForTransformer(
         }
 
         foreach ($shadowNode->getChildren() as $shadowChild) {
-            $this->traverseForTransformer($transformer, $shadowChild, $compilerContext);
+            $this->traverseForTransformer($transformer, $shadowChild, $compilerContext->withShadowTree($shadowNode));
```

Comment:
This was a 🐛, the node transformer should get the current node as shadow element.